js/models/conversations.js

```diff
@@ -169,6 +185,95 @@
+                return lookup;
+            }.bind(this)).catch(function(error) {
+                console.log(
+                    'getIdentityKeys error for conversation',
+                    this.id,
+                    error && error.stack ? error.stack : error
+                );
+                return lookup;
+            }.bind(this));
+        } else {
+            return Promise.all(this.contactCollection.map(function(contact) {
+                return textsecure.storage.protocol.loadIdentityKey(contact.id).then(function(key) {
+                    lookup[contact.id] = key;
+                }).catch(function(error) {
+                    console.log(
+                        'getIdentityKeys error for group member',
+                        contact.id,
+                        error && error.stack ? error.stack : error
+                    );
+                });
+            })).then(function() {
+                return lookup;
+            });
+        }
+    },
+    replay: function(error) {
+        var replayable = new textsecure.ReplayableError(error);
+        return replayable.replay().catch(function(error) {
+            console.log(
+                'decryptOldIncomingKeyErrors error:',
+                error && error.stack ? error.stack : error
+            );
+        });
+    },
+    decryptOldIncomingKeyErrors: function() {
+        if (this.get('decryptedOldIncomingKeyErrors')) {
+            return Promise.resolve();
+        }
+
+        var messages = this.messageCollection.filter(function(message) {
+            var errors = message.get('errors');
+            if (!errors || !errors[0]) {
+                return false;
+            }
+            var error = errors[0];
+            if (error.name !== 'IncomingIdentityKeyError') {
+                return false;
+            }
+
+            return true;
+        });
+
+        var markComplete = function() {
+            return new Promise(function(resolve) {
+                this.save({decryptedOldIncomingKeyErrors: true}).always(resolve);
+            }.bind(this));
+        }.bind(this);
+
+        if (!messages.length) {
+            return markComplete();
+        }
+
+        var safeDelete = function(message) {
+            return new Promise(function(resolve) {
+                message.destroy().always(resolve);
+            });
+        };
+
+        return this.getIdentityKeys().then(function(lookup) {
+            return Promise.all(_.map(messages, function(message) {
+                var source = message.get('source');
+                var error = message.get('errors')[0];
+                var key = lookup[source];
+                if (!key) {
+                    return;
+                }
+
+                if (equalArrayBuffers(key, error.identityKey)) {
+                    return Promise.all([this.replay(error), safeDelete(message)]);
+                }
+            }.bind(this)));
+        }.bind(this)).then(markComplete, markComplete);
```

Comment:
We catch in `getIdentityKeys()` and in `replay()` so I think we're in pretty good shape for logging.